src/content/post/2024-04-30-kuzu-v-0.4.0.md

```diff
@@ -0,0 +1,262 @@
+`/path/to/export`, utilizing the same configuration parameters as `COPY FROM` statements.
+
+```cypher
+EXPORT DATABASE TO '/path/to/export' (HEADER=true);
+```
+
+Exporting a database generates three Cypher files, one each for the schema, macro definitions and
+the `COPY` statements for each table. In addition, the data files are exported to CSV (by default),
+or Parquet, if desired.
+
+The `IMPORT DATABASE` command allows you to import the contents of
+a database from a specific directory. The query below imports the database from `/path/to/export`.
+
+```cypher
+IMPORT DATABASE FROM '/path/to/export';
+```
+
+The aim of this functionality is to make it easier to migrate databases between different
+Kùzu versions as we continue to evolve the core and the storage layer in future versions, and to
+facilitate the sharing of databases between users over longer time periods.
+
+### `COPY FROM` with subquery
+
+In v0.4.0, we added support for using subqueries following the `COPY FROM` statement. This feature allows you to
+first perform a task like `MATCH` and then use the results of that query as input to the `COPY FROM` command.
+
+For example, consider that we have a graph with a `User` node label and a `Follows` relationship type.
+We want to create a new `Person` node table and a `Knows` relationship table, where the goal is to state that
+a `Person` "knows" another `Person` if they follow each other. We can use the `COPY FROM` command with a subquery
+to achieve this as follows:
+
+```cypher
+// Create node/rel tables
+CREATE NODE TABLE Person(name STRING, PRIMARY KEY (name));
+CREATE REL TABLE Knows(FROM Person TO Person);
+// Run COPY FROM with a subquery
+COPY Person FROM (MATCH (a:User) RETURN a.name);
+COPY Knows FROM (MATCH (a:User)-[r:Follows]->(b:User) RETURN a.name, b.name);
+```
+
+An alternate use case would be when you want to directly scan data from an existing object, such as
+a Pandas DataFrame and use the results as input to the `COPY FROM` command. This can be combined with
+predicate filters as follows:
+
+```python
+# Assumes that you have a Kùzu connection object named `conn`
+# Also assumes that you created a node table named `Person` with columns `name` and `age`
+import pandas as pd
+
+df = pd.DataFrame({
+    "name": ["Adam", "Karissa", "Zhang", "Noura"],
+    "age": [30, 40, 50, 25]
+})
+
+conn.execute("COPY Person FROM (LOAD FROM df WHERE age < 30 RETURN *")
+```
+
+Using subqueries with `COPY FROM` opens up a wider range of possibilities for data manipulation and
+transformation prior to inserting data into the database.
+
+### Bulk insert into a non-empty table
+
+In prior releases, the `COPY FROM` command could only be used to bulk insert data into an empty table.
+This restriction has now been removed, and in v0.4.0, you can also bulk insert data into a non-empty table,
+making it easier to append data to an existing table. This approach is likely much more efficient than
```

Comment:
`This approach is likely much more efficient than ...`.

This claim should be correct but I would suggest not making this claim given we don't have benchmark to back it up. TBH, since we never compared against the two approaches. I cannot speak with 100% confidence it's correct.

I'll ask @ray6080 to do a quick benchmark before we actually post this claim.

Btw I would remove `likely` because we should be certain on these performance related claims.